File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
# Brute force
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dict = {}
        res = []
        for num in nums:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1
        
        
        my_dict = sorted(dict.items(), key = lambda item:item[1], reverse = True)
       
        for key,val in my_dict:
            if k == 0:
                return res
            res.append(key)
            k -= 1
        return res





            
            
            

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int num : nums){
            map.put(num, map.getOrDefault(num,0)+1);
        }
        PriorityQueue<Map.Entry<Integer,Integer>> heap = new PriorityQueue<>(Comparator.comparingInt(Map.Entry::getValue));
        for(Map.Entry<Integer, Integer> entry: map.entrySet()){
            heap.offer(entry);
            if (heap.size() > k){
                heap.poll();
            }
        }

        int[] result = new int[k];

        for(int i=0; i < k; i++){
            result[i] = heap.poll().getKey();
        }
        return result;

    }
}